core/metacling/src/TClingCallbacks.cxx

```diff
@@ -768,16 +769,15 @@ void TClingCallbacks::TransactionRollback(const Transaction &T) {
 }
 
 void TClingCallbacks::DeclDeserialized(const clang::Decl* D) {
-   if (const RecordDecl* RD = dyn_cast<RecordDecl>(D)) {
-      // FIXME: Our autoloading doesn't work (load the library) when the looked
-      // up decl is found in the PCH/PCM. We have to do that extra step, which
-      // loads the corresponding library when a decl was deserialized.
-      //
-      // Unfortunately we cannot do that with the current implementation,
-      // because the library load will pull in the header files of the library
-      // as well, even though they are in the PCH/PCM and available.
-      (void)RD;//TCling__AutoLoadCallback(RD->getNameAsString().c_str());
-   }
+  // Get module name from decl
+   clang::Module* M = D->getOwningModule();
+
+   if (!M)
+     return;
+   while (M && M->Parent)
+     M = M->Parent;
+
+   TCling__AutoLoadLibraries(M->Name);
```

Comment:
I'd prefer `TCling__AutoLoadLibraries` to take a `clang::Module*` and do there  matching the right parent module, etc. I'd suggest `TCling__AutoLoadLibraryForCxxModule`.